File: Cs50 course portfolio/Python/Readability.py
from cs50 import get_string

Text = get_string("Text: ")  # This will prompt the user to type a sentence.


def index(text):
    letters = count_letters(text)
    sentences = count_sentences(text)
    words = count_words(text)  # This part is programmed to assign the values
    

# To count the number of letters in the text
letters = 0
for i in Text:
    if (i >= "a" and i <= 'z') or (i >= "A" and i <= "Z"):
        letters += 1
        
# To count the number of sentences
sentences = 0
for i in Text:
    if i == "." or i == "!" or i == "?":
        sentences += 1
        
# To count the number of words
words = 1
for i in Text:
    if (i == " "):
        words += 1


x = (sentences * (100/words))
y = (letters * (100/words))


index = round(((0.0588 * y) - (0.296 * x) - 15.8))

if index < 1:
    print("Before Grade 1")
    
elif index > 16:
    print("Grade 16+\n")
    
else:
    print(f"Grade {index}")

File: Cs50 course portfolio/C-programming/Readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void) 
{
    string text = get_string("Text: ");
    printf("%s\n", text);
    
    int lettersCount = 0; //this is the letters count
    int wordsCount = 1; // this is the words count
    int sentencesCount = 0; // this is the sentences count
    
    for (int i = 0; i < strlen(text); i++) //counts for each
    {    
        
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            lettersCount++;
        }
        else if (text[i] == ' ')
        {
            wordsCount++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentencesCount++; //this part executes the statement
        }
    }
    
    float grade = round(0.0588 * (100 * (float) lettersCount / (float) wordsCount) - 0.296 * (100 * (float) sentencesCount /
                        (float) wordsCount) - 15.8);
    if (grade < 1)

    {
        printf("Before Grade 1\n"); // this parts takes the intput value
    } 
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }    
    else
    {
        printf("Grade %.f\n", grade);
    }
}
